fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java

```diff
@@ -364,14 +364,23 @@ private boolean needToRefreshNonPartitionTable() {
         return false;
     }
 
-    private Set<String> getPartitionsToRefreshForMaterializedView() {
+    private Set<String> getPartitionsToRefreshForMaterializedView(Map<String, String> properties)
+            throws AnalysisException {
+        String start = properties.get(TaskRun.PARTITION_START);
+        String end = properties.get(TaskRun.PARTITION_END);
+        boolean force = Boolean.parseBoolean(properties.get(TaskRun.FORCE));
         Set<String> needRefreshMvPartitionNames = Sets.newHashSet();
-
+        Set<String> rangePartitionNames = SyncPartitionUtils.getPartitionNamesByRange(materializedView, start, end);
+        LOG.debug("Materialized view refresh partitions: {}", rangePartitionNames);
+        if (force) {
+            return rangePartitionNames;
```

Comment:
if force is true, it is also necessary to go through the mapping logic of calcPotentialRefreshPartition. maybe version map should not be taken into account.

Reply:
I agree

Reply:
ok